### PR TITLE
Few touchups to the error handling during monitoring setup

### PR DESCRIPTION
**Pull request by markusthoemmes**
As per title. I mostly wanted to avoid an error like this on startup:

```
{"level":"error","ts":"2020-10-26T10:16:15.409Z","logger":"cmd","msg":"Failed to start monitoring","error":"failed to setup the Service monitor: servicemonitors.monitoring.coreos.com \"knative-openshift-metrics\" already exists","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/go/src/github.com/openshift-knative/serverless-operator/knative-operator/vendor/github.com/go-logr/zapr/zapr.go:128\nmain.main\n\t/go/src/github.com/openshift-knative/serverless-operator/knative-operator/cmd/manager/main.go:123\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:203"}
```

and ended up cleaning up a few more parts.

### CODE REVIEW

File: knative-operator/pkg/common/service_monitor.go [hunk 1]
```diff
@@ -53,28 +54,31 @@ func SetupServerlessOperatorServiceMonitor(ctx context.Context, cfg *rest.Config
 	// Create Service object to expose the metrics port(s).
 	service, err := metrics.CreateMetricsService(ctx, cfg, servicePorts)
 	if err != nil {
-		log.Info("Could not create metrics Service", "error", err.Error())
+		return fmt.Errorf("failed to create metrics service: %w", err)
```
Comment (markusthoemmes) [inline knative-operator/pkg/common/service_monitor.go:57]: @skonto did we ignore this error on purpose?

Comment (skonto) [reply knative-operator/pkg/common/service_monitor.go:57]: The idea was that monitoring will be always there on OCP and we could move on if it was not setup (I used to have code that detected if monitoring is available like the coreos packaged does).

Comment (markusthoemmes) [reply knative-operator/pkg/common/service_monitor.go:57]: Hm, sorry, not parsing that. Why'd we ignore an error during creation of the metrics service (a k8s service) and not all other errors then?

Comment (skonto) [reply knative-operator/pkg/common/service_monitor.go:57]: Wait error is returned a few lines bellow.

Comment (markusthoemmes) [reply knative-operator/pkg/common/service_monitor.go:57]: Well I didn't change anything about the flow to main, I think this purely makes the error handling a bit more "narrow" and ignores one somewhat common error that isn't really an error.

**Comment by markusthoemmes** [comment]
/retest

Infra.

File: knative-operator/pkg/common/service_monitor.go (hunk 1 above)
Comment (skonto) [reply knative-operator/pkg/common/service_monitor.go:57]: Ok makes sense but afaik there was a discussion that we should never hit this.

**Pull request merged**